compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/expression/FirCatchParameterChecker.kt

```diff
@@ -19,25 +20,30 @@ object FirCatchParameterChecker : FirTryExpressionChecker() {
     override fun check(expression: FirTryExpression, context: CheckerContext, reporter: DiagnosticReporter) {
         for (catchEntry in expression.catches) {
             val catchParameter = catchEntry.parameter
+            val source = catchParameter.source
 
             if (catchParameter.defaultValue != null) {
-                reporter.reportOn(catchParameter.source, FirErrors.CATCH_PARAMETER_WITH_DEFAULT_VALUE, context)
+                reporter.reportOn(source, FirErrors.CATCH_PARAMETER_WITH_DEFAULT_VALUE, context)
+            }
+
+            if (source != null && source.hasValOrVar()) {
```

Comment:
Nit: it's not about this PR, but rather usage/check of nullability of `source` in general (and hope this caught everyone's attention).

So... `reporter.reportOn` won't do anything if the given source is `null`. :) I remember I did add nullability checks almost everywhere in earlier stage of checkers, but recently not added anymore. Also, even in this checker, nullability of `catchParameter.source` wasn't checked at all. I'm not sure what is the convention.

Reply:
Good point, it isn't necessary and we should just continue of source is null.